api/src/test/java/org/openmrs/module/eptsreports/reporting/intergrated/library/cohorts/ResumoMensalCohortQueriesTest.java

```diff
@@ -68,6 +68,22 @@ public void getNumberOfPatientsTransferredInFromOtherHealthFacilitiesDuringCurre
     assertTrue(evaluatedCohort.getMemberIds().contains(1010));
   }
 
+  @Test
+  public void testGetNumberOfPatientsWhoInitiatedPreTarvByEndOfPreviousMonthA1()
+      throws EvaluationException {
+    CohortDefinition cd =
+        resumoMensalCohortQueries.getNumberOfPatientsWhoInitiatedPreTarvByEndOfPreviousMonthA1();
+
+    HashMap<Parameter, Object> parameters = new HashMap<>();
+    parameters.put(new Parameter("startDate", "Start Date", Date.class), this.getStartDate());
+    parameters.put(new Parameter("endDate", "End Date", Date.class), this.getEndDate());
+    parameters.put(new Parameter("location", "Location", Location.class), this.getLocation());
+
+    EvaluatedCohort evaluatedCohort = evaluateCohortDefinition(cd, parameters);
+
+    // TODO other scenarios mappings
```

Comment:
Sorry this method is not testing anything. Please implement at least 1 scenario. 
The easiest is a patient enrolled in `PRE-ART` program before `startDate`.